src/cloudflare/internal/streaming-forms.ts

```diff
@@ -0,0 +1,131 @@
+const CRLF = '\r\n';
+
+function isReadableStream(obj: unknown): obj is ReadableStream {
+  return !!(
+    obj &&
+    typeof obj === 'object' &&
+    'getReader' in obj &&
+    typeof obj.getReader === 'function'
+  );
+}
+
+function chainStreams<T>(streams: ReadableStream<T>[]): ReadableStream<T> {
+  const outputStream = new ReadableStream<T>({
+    async start(controller): Promise<void> {
+      for (const stream of streams) {
+        const reader = stream.getReader();
+
+        try {
+          // eslint-disable-next-line @typescript-eslint/no-unnecessary-condition
+          while (true) {
+            const { done, value } = await reader.read();
+            if (done) break;
+            if (value !== undefined) controller.enqueue(value);
+          }
+        } finally {
+          reader.releaseLock();
+        }
+      }
+
+      controller.close();
+    },
+  });
+
+  return outputStream;
+}
+
+type EntryOptions = { type: 'file' | 'string' };
+
+export class StreamableFormData {
+  private entries: {
+    field: string;
+    value: ReadableStream;
+    options: EntryOptions;
+  }[];
+  private boundary: string;
+
+  constructor() {
+    this.entries = [];
+
+    this.boundary = '--------------------------';
+    for (let i = 0; i < 24; i++) {
+      this.boundary += Math.floor(Math.random() * 10).toString(16);
+    }
+  }
+
+  append(
+    field: string,
+    value: ReadableStream | string,
+    options?: EntryOptions
+  ): void {
+    let valueStream: ReadableStream;
+    if (isReadableStream(value)) {
+      valueStream = value;
+    } else {
+      valueStream = new Blob([value]).stream();
+    }
+
+    this.entries.push({
+      field,
+      value: valueStream,
+      options: options || { type: 'string' },
+    });
+  }
+
+  private multipartBoundary(): ReadableStream {
+    return new Blob(['--', this.boundary, CRLF]).stream();
+  }
+
+  private multipartHeader(
+    name: string,
+    type: 'file' | 'string'
+  ): ReadableStream {
+    let filenamePart;
+
+    if (type === 'file') {
+      filenamePart = `; filename="${name}"`;
+    } else {
+      filenamePart = '';
+    }
+
+    return new Blob([
+      `content-disposition: form-data; name="${name}"${filenamePart}`,
+      CRLF,
+      CRLF,
+    ]).stream();
+  }
+
+  private multipartBody(stream: ReadableStream): ReadableStream {
```

Comment:
Can we avoid using `private` keyword, and use `#` which is in JS language?

Reply:
Might be nice to check this with the linter, looks like it can be configured: https://github.com/typescript-eslint/typescript-eslint/issues/1391#issuecomment-1124154589